bt-tests/src/test/java/bt/magnet/MagnetUriParserTest.java

```diff
@@ -53,10 +54,11 @@ public void testParser_v1_AllParamsPresent() {
         assertTrue(trackerUrls.contains(trackerUrl2));
 
         Collection<InetPeerAddress> peerAddresses = uri.getPeerAddresses();
-        assertEquals(2, peerAddresses.size());
-        Iterator<InetPeerAddress> iter = peerAddresses.iterator();
-        assertEquals(new InetPeerAddress("1.1.1.1", 10000), iter.next());
-        assertEquals(new InetPeerAddress("2.2.2.2", 10000), iter.next());
+        Collection<InetPeerAddress> expectedPeers = Set.of(
+            new InetPeerAddress("1.1.1.1", 10000),
+            new InetPeerAddress("2.2.2.2", 10000)
+        );
+        assertEquals(expectedPeers, peerAddresses);
```

Comment:
Better to wrap expectedPeers in a HashSet - rather than assume that it is a set.
`assertEquals(new HashSet<>(expectedPeers), peerAddresses);`